accepted/00000-ptf-repos-in-uyuni.md

```diff
@@ -0,0 +1,101 @@
+The resulting packages are the last released maintenance update plus a patch which fixes the actual issue.
+
+PTFs are made for dedicated customers. This means every customer has different PTF repositories, with different content and fixes for different service requests.
+
+With Uyuni we should provide the possibility to mirror the content of these repositories and let the customer use it for there clients.
+
+
+# Detailed design
+[design]: #detailed-design
+
+The mapping of SCC repositories to Uyuni Channels is done via a prepared product tree file which contains the definitions for all products and its repositories.
+It enhances the information from SCC with values required for Uyuni.
+
+Such an entry is defined by 3 values:
+
+- the product id
+- the product id of the root product
+- the repository id
+
+The following values are set by the product tree:
+
+- channel label: it must be unique on the whole Uyuni Server
+- parent channel label: what is the parent of this channel
+- channel name: similar to channel label but human readable
+- update tag: a special value to modify the errata id during import
+- mandatory: defines if this channel is required for the product it belongs to
+
+As PTF repositories are customer specific, such a hard coded definition cannot 
+be provided upfront.
+They must be generated out of existing data.
+
+As SCC needs to crawl the buildservice to find all the existing PTF repositories, they agreed on a path structure to map the repositories to the products.
+The defined path is:
+
+    PTF/Release/<Customer Account ID>/<Product Identifier>/<Product Version>/<Product Architecture>/ptf[_debug]/
+
+When Uyuni is listing the repositories available for an organizations, it may find repositories which `id` is not known in the complete product tree.
+To identify them as PTF repositories we can check if the path start with `PTF/Release`.
+
+If they are PTF repositories, we need to generate the missing values for the Channel mapping like this:
+
+- repository id: take it from the repository definition
+- product id: search for the product using `Product Identifier`, `Product Version`, and `Product Architecture` like specified in the URL
+- root product id: find all existing root products for the product id. We need to generate an entry for all of them
+
+The values for update tag and mandatory can be hard coded as they are the same 
+for all PTF repositories.
+
+- update tag: empty
+- mandatory: false
+
+As we need to generate entries for all existing root products we can iterate over all we found.
+For every root product we take the first channel we can find which does have a `parent channel label` set.
+We can take this as parent channel label for our new entry.
+
+As the channel label and name must be unique.
+We generate the label and name from the URL components:
+
+    <Customer Account ID> <Product Identifier> <Product Version> "PTFs" ["Debuginfo"] <Product Architecture>
```

Comment:
How about we use that name already in SCC for PTF repos, and you just take it from the API response? Then those repositories would have the same name in SCC, RMT and SUSE Manager.

Reply:
I think this is handled in the alternative below - with downside of parsing the full tree per organization (mirror/credentials)

Reply:
This would help a lot. In this case we only need to append the suffix when we duplicate it for different root products.

Reply:
@digitaltom I would like this RFC to be concluded whenever possible. Can we assume the generation of this label is on SCC and it will be delivered in the API at this point?

Otherwise I'd suggest adapting the text below to make clear the label generation is on Uyuni, and proceed under that assumption for now. Then, if needed, handle a change at a later point.

Reply:
For now SUSE Manager will generate the names. We can change it later if wanted.